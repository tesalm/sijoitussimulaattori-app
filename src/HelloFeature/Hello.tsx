import React from 'react';
import { Platform, Text, View, Button } from 'react-native';
import CounterView from './components/CounterView';
import { helloStyles as styles } from './styles';
import { RootState } from '../redux/reducers';
import { Dispatch, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { CounterAction, addToCounter, removeFromCounter } from './actions';

const instructions = Platform.select({
  ios: 'You are on iOS',
  android: 'You are on Android',
});

export interface HelloProps {
  name?: string;
  counterValue: number;
  onIncrement: typeof addToCounter;
  onDecrement: typeof removeFromCounter;
}

export class Hello extends React.Component<HelloProps> {
  constructor(props: HelloProps) {
    super(props);
  }

  render() {
    const { name, counterValue, onIncrement, onDecrement } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Hello, {name || 'World'}!</Text>
        <Text style={styles.instructions}>{instructions}</Text>
        <CounterView counterValue={counterValue} />
        <Button
          onPress={() => onIncrement && onIncrement(1)}
          title="Increment counter"
        />
        <Button onPress={onDecrement} title="Decrement counter" />
      </View>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  counterValue: state.counter.counterValue,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      onIncrement: addToCounter,
      onDecrement: removeFromCounter,
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Hello);
