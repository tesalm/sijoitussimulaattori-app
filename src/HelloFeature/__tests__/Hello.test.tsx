import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Hello, HelloProps } from '../Hello';

describe('Hello World', () => {
  const defaultHelloProps: HelloProps = {
    counterValue: 0,
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };

  it('renders correctly', async () => {
    const component = renderer
      .create(<Hello {...defaultHelloProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('has property name correctly set', async () => {
    const wrapper = mount(<Hello name="John" {...defaultHelloProps} />);
    expect(wrapper.prop('name')).toEqual('John');
  });
});
