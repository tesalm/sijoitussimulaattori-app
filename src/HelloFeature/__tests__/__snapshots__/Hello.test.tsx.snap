// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    World
    !
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    You are on iOS
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 1,
        "textAlign": "center",
      }
    }
  >
    Counter value is:
  </Text>
  <Text
    style={
      Object {
        "fontSize": 40,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment counter
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement counter
      </Text>
    </View>
  </View>
</View>
`;
